lib/inputstreamhelper/__init__.py

```diff
@@ -540,6 +549,37 @@ def _chromeos_config(self):
 
         return devices
 
+    def _remove_old_backups(self, backup_path):
+        """Removes old Widevine backups, if number of allowed backups is exceeded"""
+        backups = int(get_setting('backups', '4'))
+        versions = sorted([LooseVersion(version) for version in os.listdir(backup_path)])
+
+        if 'x86' in self._arch():
+            installed_version = self._load_widevine_config()['version']
+        else:
+            installed_version = self._select_best_chromeos_image(self._load_widevine_config())['version']
+
+        while len(versions) > backups + 1:
+            remove_version = str(versions[0])
+            if remove_version == installed_version:
+                try:
+                    remove_version = str(versions[1])
+                except IndexError:
+                    break
+
+            log('removing oldest backup which is not installed: {version}', version=remove_version)
+            shutil.rmtree(os.path.join(backup_path, remove_version))
+            versions = sorted([LooseVersion(version) for version in os.listdir(backup_path)])
+
+    def _install_cdm_from_backup(self, version):
+        """Copies files from specified backup version to cdm dir"""
+        filenames = os.listdir(os.path.join(self._backup_path(), version))
+
+        for filename in filenames:
+            xbmcvfs.copy(os.path.join(self._backup_path(), version, filename), os.path.join(self._ia_cdm_path(), filename))
```

Comment:
As I said multiple times now, we ought to use hard links where they are supported. And fall back to copying when it is not. Otherwise we may be breaking a working setup because the disk is full in cases where the fix already works fine.

We don't fix bugs on a basis of lowest common denominator.

Reply:
I really don't understand your obsession on those links (which are not used anywhere at the moment, so I'm not sure which fix you are talkling about). If your system ever has less than 10MB space left, you have more serious problems. I just wanted to keep the OS specific part to a minimum, otherwise we could just write a different addon for each system.

That being said, I'll add those links, since this is only a small change and not worth the whole discussion.

Reply:
It is not an obsession, it's avoiding potential known problems. Why not do that when it works on most platforms. That said, if you don't want to do it, I don't mind adding it later.